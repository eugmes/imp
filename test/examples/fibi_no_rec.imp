function fibi(x : integer) return integer is
  var a, b, c : integer;
  var i : integer;
begin
  a = 1;
  b = 1;
  c = 0;
  i = 3;

  while i <= x loop
    c = a + b;
    a = b;
    b = c;
    i = i + 1;
  end loop;

  return b;
end fibi;

procedure main () is
  var n : integer;
begin
  output("Enter number: "); input(n);
  output("Result is: "); output(fibi(n));
  newline;
end main;
