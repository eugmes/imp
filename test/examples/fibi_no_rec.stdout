Enter number: Result is: 1836311903
